internal/gotty/client.go

```diff
@@ -0,0 +1,208 @@
+package gotty
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/url"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/containerd/console"
+	"github.com/gorilla/websocket"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+// Scaleway GoTTY websocket endpoint
+var wsURLs = map[scw.Zone]string{
+	scw.ZoneFrPar1: "wss://tty-par1.scaleway.com/v2/ws",
+	scw.ZoneFrPar2: "wss://tty-par2.scaleway.com/v2/ws",
+	scw.ZoneNlAms1: "wss://tty-ams1.scaleway.com/v2/ws",
+}
+
+const (
+	// GoTTY ingress message code
+	outputCode         = '0'
+	setWindowTitleCode = '2'
+
+	// GoTTY egress message code
+	inputCode          = '0'
+	pingCode           = '1'
+	resizeTerminalCode = '2'
+)
+
+type Client struct {
+	wsURL     string
+	serverId  string
+	secretKey string
+}
+
+// NewClient returns a GoTTY client.
+func NewClient(zone scw.Zone, serverID string, secretKey string) (*Client, error) {
+
+	wsURL, zoneExist := wsURLs[zone]
+	if !zoneExist {
+		return nil, fmt.Errorf("gotty is not availabe in zone %s", zone)
+	}
+
+	return &Client{
+		wsURL:     wsURL,
+		serverId:  serverID,
+		secretKey: secretKey,
+	}, nil
+}
+
+func (c *Client) Connect() error {
+
+	wsDialer := websocket.Dialer{}
+	conn, _, err := wsDialer.Dial(c.wsURL, nil)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		// Websocket protocol require the server to close the connection.
+		// This sent a close request.
+		conn.WriteMessage(websocket.CloseMessage, websocket.FormatCloseMessage(websocket.CloseNormalClosure, ""))
+	}()
+
+	// This is how scaleway implement gotty authentication
+	err = conn.WriteJSON(map[string]string{
+		"AuthToken": "",
+		"Arguments": "?" + url.Values{"arg": []string{c.secretKey, c.serverId}}.Encode(),
+	})
+	if err != nil {
+		return err
+	}
+
+	cns, err := console.ConsoleFromFile(os.Stdin)
+	if err != nil {
+		return fmt.Errorf("os.Stdin doesn't seems to be a valid terminal: %w", err)
+	}
+	err = cns.SetRaw()
+	if err != nil {
+		return fmt.Errorf("error setting raw terminal: %w", err)
+	}
+	defer cns.Reset()
+	defer cns.Close()
+
+	// Create a chanel that will receive all resizes signals
+	resizeChan := make(chan os.Signal, 1)
+	signal.Notify(resizeChan, syscall.SIGWINCH)
```

Comment:
We should split the syscall in another `_unix.go` file and a dummy one in a `_windows.go` file with the righ build tags